src/vector_index.h

```diff
@@ -31,6 +31,7 @@
 
 typedef enum {
   VECSIM_QT_KNN,
+  VECSIM_QT_RANGE
```

Comment:
nitpicking - need to handle the new type in `VectorQuery_Free`, `QueryNode_DumpSds`, or others?

Reply:
Good question - after checking, it looks like to changes are required there